### PR TITLE
Calling abort inside TestServer handler should set RequestAborted

### PR DESCRIPTION
**Pull request by huysentruitw**
Calling abort inside TestServer handler should set RequestAborted

Addresses #7449

### CODE REVIEW

File: src/Hosting/TestHost/src/RequestLifetimeFeature.cs
Comment (JamesNK) [inline src/Hosting/TestHost/src/RequestLifetimeFeature.cs]: You can support with set like this: https://github.com/grpc/grpc-dotnet/blob/0be935fca79674be56c9fe7e69f6708306889ae2/testassets/FunctionalTestsWebsite/Infrastructure/TestHttpRequestLifetimeFeature.cs

**Review by JamesNK — COMMENTED** [review]
Looks ok to me. Should be reviewed by a TestHost Expert(tm)

File: src/Hosting/TestHost/src/RequestLifetimeFeature.cs
Comment (huysentruitw) [reply src/Hosting/TestHost/src/RequestLifetimeFeature.cs]: That's lying to the API user though. That setter should not exist IMO as setting it breaks the Abort functionality.

File: src/Hosting/TestHost/src/RequestLifetimeFeature.cs
Comment (Tratcher) [inline src/Hosting/TestHost/src/RequestLifetimeFeature.cs]:
```suggestion
namespace Microsoft.AspNetCore.TestHost
```

File: src/Hosting/TestHost/src/RequestLifetimeFeature.cs
Comment (Tratcher) [inline src/Hosting/TestHost/src/RequestLifetimeFeature.cs]: It's the caller's responsibility to link tokens, and they may choose to link them in other ways. You can replace _linkedCancellationTokenSource with a field for the current CancellationToken initialized to _abortableCancellationTokenSource.Token.

```suggestion
        public CancellationToken RequestAborted { get; set; } = _abortableCancellationTokenSource.Token;
```

**Review by Tratcher — CHANGES_REQUESTED** [review]

File: src/Hosting/TestHost/src/RequestLifetimeFeature.cs
Comment (huysentruitw) [reply src/Hosting/TestHost/src/RequestLifetimeFeature.cs]: So, when the caller passes a `CancellationToken`, the `Abort()` method stops working. That's very weird to me 🤔

Comment (Tratcher) [reply src/Hosting/TestHost/src/RequestLifetimeFeature.cs]: Yes, that's the behavior of all of our servers (though Abort also does other things in most servers like close the socket). The caller setting CancellationToken is intentionally customizing the behavior, they can choose how the linking happens.

**Review by JamesNK — APPROVED** [review]

**Comment by JamesNK** [comment]
Waiting on @Tratcher approval before merging.

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**
